tutorials/dlp-hybrid-inspect/src/main/java/com/example/dlp/HybridInspectSql.java

```diff
@@ -63,104 +64,215 @@
-        cmd.getOptionValue(databaseUser.getOpt()),
-        databasePassword,
-        databaseType,
-        Integer.parseInt(cmd.getOptionValue(sampleRowLimit.getOpt())),
-        cmd.getOptionValue(hybridJobName.getOpt()),
-        Integer.parseInt(cmd.getOptionValue(threadPoolSize.getOpt(), "4")));
+
+    java.util.Date endDate = new java.util.Date();
+    long diff = endDate.getTime() - startDate.getTime();
+    long diffSeconds = diff / 1000 % 60;
+    long diffMinutes = diff / (60 * 1000) % 60;
+    long diffHours = diff / (60 * 60 * 1000) % 24;
+    long diffDays = diff / (24 * 60 * 60 * 1000);
+
+    System.out.println();
+    System.out.println("-----------------------------------------------------------------");
+    System.out.println(" End Run ID: " + runId);
+    System.out.println(" Total Databases scanned: " + totalDatabaseCount);
+    System.out.println(" Total Tables scanned: " + totalTableCount);
+    System.out.println(" End Time: " + endDate.toString());
+    System.out.print  (" Duration: ");
+    System.out.print  ("  " + diffDays + " days, ");
+    System.out.print  ("  " + diffHours + " hours, ");
+    System.out.print  ("  " + diffMinutes + " minutes, ");
+    System.out.println("  " + diffSeconds + " seconds.");
+    System.out.println("-----------------------------------------------------------------");
+
   }
 
   /**
    * This method uses JDBC to read data from a SQL database, then chunks it and sends it to a Cloud
    * DLP Hybrid Job
    */
-  private static void inspectSQLDb(
-      String databaseInstanceServer,
-      String databaseInstanceDescription,
-      String databaseName,
-      String tableName,
-      String databaseUser,
-      String databasePassword,
-      String databaseType,
-      int sampleRowLimit,
+  private static Integer inspectSQLDb(
+      Database database,
       String hybridJobName,
-      int threadPoolSize) {
-    // generate a UUID as a tracking number for the scan job. This is sent as a label in the Hybrid
-    // Inspect Request and used for tracking.
-    String runId = UUID.randomUUID().toString();
+      String hybridJobTriggerName,
+      String runId) {
 
-    System.out.println("-----------------------------------------");
-    System.out.println(" Start Run ID: " + runId);
-    System.out.println("-----------------------------------------");
+    System.out.println();
+    System.out.print(String.format(">> [%s,%s:%s]: Starting Inspection", database.databaseInstanceDescription, database.databaseInstanceServer, database.databaseName));
+
+    //retreive the password from Secret Manager
+    String databasePassword_pre = null;
+    LOG.info(String.format(">> Retrieving password from Secret Manager (%s)",
+        database.secretManagerResourceName));
+    try {
+      databasePassword_pre = accessSecretVersion(ServiceOptions.getDefaultProjectId(),
+          database.secretManagerResourceName,"latest");
+    } catch (Exception ex) {
+      ex.printStackTrace();
+    }
```

Comment:
Avoid doing this: better to just leave the exception uncaught so it propagates to the calling code. When you do a get on your future, check for ExecutionException and log that there was an error processing database *foo*.

Reply:
fixing.